crates/engine_provider/src/engine.rs

```diff
@@ -98,6 +98,16 @@ pub trait EngineApi<N: NetworkAttributes>: Send + Sync {
         payload_attributes: Option<N::PayloadAttributes>,
     ) -> TransportResult<ForkchoiceUpdated>;
 
+    /// Updates the execution layer client with the given fork choice, as specified for the Cancun
+    /// fork.
+    ///
+    /// See also <https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#engine_forkchoiceupdatedv3>
+    async fn fork_choice_updated_v4(
```

Comment:
this method actually doesn't exist yet -- per the EIP, only [getPayloadV4](https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md#engine_getpayloadv4) and [newPayloadV4](https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md#engine_newpayloadv4) were added